api/server/router/image/image_routes.go

```diff
@@ -247,8 +248,17 @@ func (s *imageRouter) getImagesJSON(ctx context.Context, w http.ResponseWriter,
 		return err
 	}
 
-	// FIXME: The filter parameter could just be a match filter
-	images, err := s.backend.Images(r.Form.Get("filters"), r.Form.Get("filter"), httputils.BoolValue(r, "all"), false)
+	imageFilters, err := filters.FromParam(r.Form.Get("filters"))
+	if err != nil {
+		return err
+	}
+
+	version := httputils.VersionFromContext(ctx)
+	if versions.LessThan(version, "1.28") && r.Form.Get("filter") != "" {
```

Comment:
Minor minor nit, but store the value of the filter field instead of getting it twice.